pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -19,24 +19,33 @@ package tektonpipeline
 import (
 	"context"
 	"fmt"
-	"os"
-	"path/filepath"
+	"reflect"
+	"time"
 
 	mf "github.com/manifestival/manifestival"
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
 	clientset "github.com/tektoncd/operator/pkg/client/clientset/versioned"
 	tektonpipelinereconciler "github.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektonpipeline"
 	"github.com/tektoncd/operator/pkg/reconciler/common"
+	"github.com/tektoncd/operator/pkg/reconciler/installer"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/apis"
 	"knative.dev/pkg/logging"
 	pkgreconciler "knative.dev/pkg/reconciler"
 )
 
-// Pipelines ConfigMap
 const (
+	// Pipelines ConfigMap
 	featureFlag    = "feature-flags"
 	configDefaults = "config-defaults"
+
+	proxyLabel = "operator.tekton.dev/disable-proxy=true"
+)
+
+var (
+	lastAppliedManifest mf.Manifest
```

Comment:
let us avoid this completely. This means we are storing state information related to an installtion inside a controller.
This might look ok in our case as we can assume, there will be only one instance of our CRD. However, this could be very confusing for anyone new reading this code as this essentially is storing state in a stateless component (controller).

Reply:
i have added comments on 2 different places in this file where `lastAppliedManifest` could be replaced/eliminated.